libraries/ESP8266WiFiMesh/src/TcpIpMeshBackend.cpp

```diff
@@ -0,0 +1,560 @@
+
+void TcpIpMeshBackend::setAPModeTimeout(uint32_t apModeTimeoutMs)
+{
+  _apModeTimeoutMs = apModeTimeoutMs;
+}
+
+uint32_t TcpIpMeshBackend::getAPModeTimeout() {return _apModeTimeoutMs;}
+
+/**
+ * Disconnect completely from a network.
+ */
+void TcpIpMeshBackend::fullStop(WiFiClient &currClient)
+{
+  currClient.stop();
+  yield();
+  WiFi.disconnect();
+  yield();
+}
+
+/**
+ * Wait for a WiFiClient to transmit
+ *
+ * @return True if the client is ready, false otherwise.
+ * 
+ */
+bool TcpIpMeshBackend::waitForClientTransmission(WiFiClient &currClient, uint32_t maxWait)
+{
+  uint32_t connectionStartTime = millis();
+  uint32_t waitingTime = millis() - connectionStartTime;
+  while(currClient.connected() && !currClient.available() && waitingTime < maxWait)
+  {
+    delay(1);
+    waitingTime = millis() - connectionStartTime;
+  }
+
+  /* Return false if the client isn't ready to communicate */
+  if (WiFi.status() == WL_DISCONNECTED && !currClient.available())
+  {
+    verboseModePrint(F("Disconnected!")); 
+    return false;
+  }
+  
+  return true;
+}
+
+/**
+ * Send the mesh instance's current message then read back the other node's response
+ * and pass that to the user-supplied responseHandler.
+ *
+ * @param currClient The client to which the message should be transmitted.
+ * @return A status code based on the outcome of the exchange.
+ * 
+ */
+transmission_status_t TcpIpMeshBackend::exchangeInfo(WiFiClient &currClient)
+{
+  verboseModePrint("Transmitting");  // Not storing strings in flash (via F()) to avoid performance impacts when using the string.
+    
+  currClient.print(getCurrentMessage() + '\r');
+  yield();
+
+  if (!waitForClientTransmission(currClient, _stationModeTimeoutMs))
+  {
+    fullStop(currClient);
+    return TS_CONNECTION_FAILED;
+  }
+
+  if (!currClient.available()) 
+  {
+    verboseModePrint(F("No response!"));
+    return TS_TRANSMISSION_FAILED; // WiFi.status() != WL_DISCONNECTED so we do not want to use fullStop(currClient) here since that would force the node to scan for WiFi networks.
+  }
+
+  String response = currClient.readStringUntil('\r');
+  yield();
+  currClient.flush();
+
+  /* Pass data to user callback */
+  return getResponseHandler()(response, *this);
+}
+
+/**
+ * Handle data transfer process with a connected AP.
+ *
+ * @return A status code based on the outcome of the data transfer attempt.
+ */
+transmission_status_t TcpIpMeshBackend::attemptDataTransfer()
+{
+  // Unlike WiFi.mode(WIFI_AP);, WiFi.mode(WIFI_AP_STA); allows us to stay connected to the AP we connected to in STA mode, at the same time as we can receive connections from other stations. 
+  // We cannot send data to the AP in STA_AP mode though, that requires STA mode. 
+  // Switching to STA mode will disconnect all stations connected to the node AP (though they can request a reconnect even while we are in STA mode).
+  WiFiMode_t storedWiFiMode = WiFi.getMode();
+  WiFi.mode(WIFI_STA);
+  delay(1);
```

Comment:
Interesting delay here and below (ref #6630 )

Reply:
This could actually be directly related to #6630, not completely sure why the delays were added.